Sources/VLCRendererDiscovererManager.swift

```diff
@@ -0,0 +1,257 @@
+
+    @objc var rendererButtons: [UIButton] = [UIButton]()
+
+    fileprivate init(presentingViewController: UIViewController?) {
+        self.presentingViewController = presentingViewController
+        super.init()
+    }
+
+    // Returns renderers of *all* discoverers
+    @objc func getAllRenderers() -> [VLCRendererItem] {
+        var renderers = [VLCRendererItem]()
+
+        for discoverer in discoverers {
+            renderers += discoverer.renderers
+        }
+        return renderers
+    }
+
+    fileprivate func isDuplicateDiscoverer(with description: VLCRendererDiscovererDescription) -> Bool {
+        for discoverer in discoverers {
+            if discoverer.name == description.name {
+                return true
+            }
+        }
+        return false
+    }
+
+    @discardableResult @objc func start() -> Bool {
+        // Gather potential renderer discoverers
+        guard let tmpDiscoverers: [VLCRendererDiscovererDescription] = VLCRendererDiscoverer.list() else {
+            return false
+        }
+        for discoverer in tmpDiscoverers {
+            if !isDuplicateDiscoverer(with: discoverer) {
+                if let rendererDiscoverer = VLCRendererDiscoverer(name: discoverer.name) {
+                    if rendererDiscoverer.start() {
+                        rendererDiscoverer.delegate = self
+                        discoverers.append(rendererDiscoverer)
+                    } else {
+                        print("Unable to start renderer discoverer with name: \(rendererDiscoverer.name)")
+                    }
+                } else {
+                    print("Unable to instanciate renderer discoverer with name: \(discoverer.name)")
+                }
+            }
+        }
+
+        return true
+    }
+
+    @objc func stop() {
+        for discoverer in discoverers {
+            discoverer.stop()
+        }
+        discoverers.removeAll()
+    }
+
+    // MARK: VLCActionSheet
+    @objc fileprivate func displayActionSheet() {
+        if let presentingViewController = presentingViewController {
+            // If only one renderer, choose it automatically
+            if getAllRenderers().count == 1 {
+                let indexPath = IndexPath(row: 0, section: 0)
+                if let rendererItem = getAllRenderers().first {
+                    actionSheet.collectionView.selectItem(at: indexPath, animated: false, scrollPosition: .centeredVertically)
+                    actionSheet(collectionView: actionSheet.collectionView, didSelectItem: rendererItem, At: indexPath)
+                    setRendererItem(rendererItem: rendererItem)
+
+                    if let movieViewController = presentingViewController as? VLCMovieViewController {
+                        movieViewController.setupCastWithCurrentRenderer()
+                    }
+                }
+            } else {
+                presentingViewController.present(actionSheet, animated: false, completion: nil)
```

Comment:
why animated false ?

Reply:
Because the animation is handled by `VLCActionSheet`.